packages/back-end/src/models/ExperimentModel.ts

```diff
@@ -98,7 +106,292 @@ const experimentSchema = new mongoose.Schema({
   ideaSource: String,
 });
 
-export const ExperimentModel = mongoose.model<ExperimentDocument>(
+type ExperimentDocument = mongoose.Document & ExperimentInterface;
+
+const ExperimentModel = mongoose.model<ExperimentDocument>(
   "Experiment",
   experimentSchema
 );
+
+/**
+ * Convert the Mongo document to an ExperimentInterface, omitting Mongo default fields __v, _id
+ * @param doc
+ */
+const toInterface = (doc: ExperimentDocument): ExperimentInterface =>
+  omit(doc.toJSON(), ["__v", "_id"]);
+
+export async function getExperimentById(
+  id: string
+): Promise<ExperimentInterface | null> {
+  const experiment = await ExperimentModel.findOne({
+    id,
+  });
+  return experiment ? toInterface(experiment) : null;
+}
+
+export async function getAllExperiments(
+  organization: string,
+  project?: string
+): Promise<ExperimentInterface[]> {
+  const query: FilterQuery<ExperimentDocument> = {
+    organization,
+  };
+
+  if (project) {
+    query.project = project;
+  }
+
+  return (await ExperimentModel.find(query)).map((m) => toInterface(m));
+}
+
+export async function getExperimentByTrackingKey(
+  organization: string,
+  trackingKey: string
+): Promise<ExperimentInterface | null> {
+  const experiment = await ExperimentModel.findOne({
+    organization,
+    trackingKey,
+  });
+
+  return experiment ? toInterface(experiment) : null;
+}
+
+export async function getExperimentsByIds(
+  ids: string[]
+): Promise<ExperimentInterface[]> {
+  const experiments = await ExperimentModel.find({
+    id: { $in: ids },
+  });
+
+  return experiments.map((m) => toInterface(m));
+}
+
+export async function deleteExperimentById(id: string): Promise<void> {
+  await ExperimentModel.deleteOne({
+    id,
+  });
+}
+
+export async function getExperimentsUsingSegment(
+  id: string,
+  orgId: string
+): Promise<ExperimentInterface[]> {
+  const experiments = await ExperimentModel.find(
+    {
+      organization: orgId,
+      segment: id,
+    },
+    { id: 1, name: 1 }
+  );
+
+  return experiments.map((m) => toInterface(m));
+}
+
+export async function getSampleExperiment(
+  organization: string
+): Promise<ExperimentInterface | null> {
+  const exp = await ExperimentModel.findOne({
+    organization,
+    id: /^exp_sample_/,
+  });
+
+  return exp ? toInterface(exp) : null;
+}
+
+export async function createExperiment(
+  data: Partial<ExperimentInterface>,
+  organization: OrganizationInterface
+): Promise<ExperimentInterface> {
+  if (!data.organization) {
+    throw new Error("Missing organization");
+  }
+  if (data.organization !== organization.id) {
+    throw new Error("Experiment and Organization must match");
+  }
```

Comment:
Instead of this, we could just set `data.organization = organization.id` to force it to always match.  That way, the calling function doesn't even need to specify `experiment.organization`, it will be added for them.